scopes/dependencies/dependency-resolver/dependency-version-resolver.ts

```diff
@@ -29,7 +36,7 @@ export class DependencyVersionResolver {
       options
     );
     // TODO: the cache should be probably passed to the package manager constructor not to the install function
-    const resolved = await this.packageManager.resolveRemoteVersion(packageName, calculatedOpts);
+    const resolved = this.limitRequests(() => this.packageManager.resolveRemoteVersion(packageName, calculatedOpts));
```

Comment:
With this solution, using `p-limit` in DependencyVersionResolver, resolving remote versions will always be limited, so an overload will not happen wherever the calls are coming from.

I can rewrite it to using `p-map` with a set concurrency in the map where each package's latest version is resolved: 

https://github.com/teambit/bit/blob/5d6af2326358600db819d550e864c5d40551d907/scopes/dependencies/dependency-resolver/dependency-resolver.main.runtime.ts#L976-L984

Do you prefer the `p-map` solution? @GiladShoham @davidfirst

Reply:
As discussed, let's keep the `p-limit`, it makes sense to have it here.